### ISSUE TITLE
Unhelpful error message when attempting to re-run failed create

### ISSUE THREAD

**Issue by robdimsdale** [pivotal-cf-experimental/bosh-bootloader#10]
```
± |master S:3 ✗| → bbl --aws-access-key-id=$(cat ~/workspace/deployments-rmd/bbl-concourse/aws_access_key_id) --aws-secret-access-key=$(cat ~/workspace/deployments-rmd/bbl-concourse/aws_secret_access_key) --state-dir="~/workspace/deployments-rmd/bbl-concourse/state" --aws-region=us-east-1 unsupported-deploy-bosh-on-aws-for-concourse
step: creating keypair
step: generating cloudformation template
step: creating cloudformation stack
•

aws cloudformation failed: ROLLBACK_COMPLETE
```

What do I do with this error? I tried to `destroy` but that didn't work:

```
± |master S:3 ✗| → bbl --aws-access-key-id=$(cat ~/workspace/deployments-rmd/bbl-concourse/aws_access_key_id) --aws-secret-access-key=$(cat ~/workspace/deployments-rmd/bbl-concourse/aws_secret_access_key) --state-dir="~/workspace/deployments-rmd/bbl-concourse/state" --aws-region=us-east-1 destroy
Are you sure you want to delete your infrastructure? This operation cannot be undone! (y/N): y
step: destroying infrastructure


stack not found
```

I deleted the Cloud Formation stack manually in the UI and this seems to allow me to progress.

**Comment by robdimsdale** [comment]
I seem to get this failure even when the step that fails is after the cloudformation step (e.g. the bosh-init deploy fails). Every time I have to delete the cloudformation stack manually in order to run the tool again.

**Comment by zankich** [comment]
@robdimsdale one current limitation of bbl, is that we only write out the `state.json` when the command successfully finishes, so when your command failed, it left AWS artifacts which are not tracked by bbl. Also, the `destroy` command needs stack and bosh information in order to do it's job, so your `destroy` failed because that information didn't exist.

We have a track of stories for bbl to gracefully handle errors during a provision, but until we get to that point, it may be worth providing a more useful error message letting the user know that they should delete the stack which bbl created.

**Comment by Amit-PivotalLabs** [comment]
Hey @robdimsdale 

Do you have any events from the AWS console explaining why the original cloudformation stack might have failed?

We have stories that would fix the `destroy` problem you hit.  But we should still improve the original error message.

Instead of:

```
aws cloudformation failed: ROLLBACK_COMPLETE
```

It should say:

```
Failed creating cloudformation stack `STACK_NAME', check the AWS console for error events related to this stack, and/or open a GitHub issue at https://github.com/pivotal-cf-experimental/bosh-bootloader/issues.
```

**Comment by robdimsdale** [comment]
No logs anymore, but here's what I think happened:

- run `bbl` first time - bosh-init step fails due to nokogiri / xcode
- install xcode and nokogiri, re-run `bbl` - I'm not quite sure what happened here, but I think `bbl` had no state file so tried to create an entirely new stack. This failed because I reached my VPC limit. I remember seeing two stacks in Cloudformation.
- I deleted both stacks (including having to destroy the bosh-init director manually from the first stack as it wasn't part of Cloud Formation) and ran `bbl` again and it succeeded.
